internal/controller/gateway_mutator.go

```diff
@@ -389,3 +393,73 @@ func (g *gatewayMutator) mutatePod(ctx context.Context, pod *corev1.Pod, gateway
 	}
 	return nil
 }
+
+// fetchGatewayAndConfig returns the Gateway and the referenced GatewayConfig (if present).
+func (g *gatewayMutator) fetchGatewayAndConfig(ctx context.Context, gatewayName, gatewayNamespace string) (*gwapiv1.Gateway, *aigv1a1.GatewayConfig) {
+	// Fetch the Gateway object.
+	var gateway gwapiv1.Gateway
+	if err := g.c.Get(ctx, client.ObjectKey{Name: gatewayName, Namespace: gatewayNamespace}, &gateway); err != nil {
+		if apierrors.IsNotFound(err) {
+			g.logger.Info("Gateway not found, using global defaults",
```

Comment:
I read like "default of what?" so let's be explicit about it
```suggestion
			g.logger.Info("Gateway not found, using global default configuration",
```